packages/liveblocks-core/src/lib/fsm.ts

```diff
@@ -538,8 +588,12 @@ export class FSM<
       nextState = nextTarget;
     } else {
       nextState = nextTarget.target;
-      assign = nextTarget.assign;
-      effect = nextTarget.effect;
+      effects =
+        nextTarget.effect !== undefined
```

Comment:
I love concise code but setting something to undefined if it's undefined because of nested ternaries may be a sign to just not use a double ternary 😂 

```
if (nextTarget.effect !== undefined) {
 effects = Array.isArray(nextTarget.effect)
            ? nextTarget.effect
            : [nextTarget.effect];
}
```

Would be clearer here I think.

Reply:
Better even, because `effect` can no longer be undefined, we could remove it entirely 👍 Fixed in 84d6d2e229b0e542d06a91db59efad8bce20f4a8.